app/javascript/components/bathroom.jsx

```diff
@@ -1,16 +1,14 @@
 import React from 'react';
 import PropTypes from 'prop-types';
-import { Query } from 'react-apollo';
 import gql from 'graphql-tag';
 import styled from 'styled-components';
 import { WhiteSubTitle, WhiteTitleLarge } from '@components/typography';
 import { LoadingMessage, ErrorMessage } from '@messages/default-messages';
+import withFragment from './hocs/with-fragment';
 
-const getBathroomCode = gql`
-  query getBathroomCode($cityName: String!) {
-    location(cityName: $cityName) {
-      bathroomCode
-    }
+export const getBathroomCode = gql`
+  fragment Bathroom on Location {
+    bathroomCode
```

Comment:
Why doesn't the fragment need parameters (i.e. $cityName: String!)?

Reply:
The only requirement for Bathroom (and the other components) is that this data come from a `Location` type. That could be composed in any Location type result but in our case is composed in the `getWidgets` query that specifies the city_name for a single location.